include/parallel/parallel_implementation.h

```diff
@@ -2052,6 +2049,29 @@ inline status Communicator::probe (const unsigned int src_processor_id,
   return stat;
 }
 
+template<typename T>
+inline Status Communicator::packed_range_probe (const unsigned int src_processor_id,
+                                                const MessageTag & tag,
+                                                bool & flag) const
+{
+  START_LOG("packed_range_probe()", "Parallel");
+
+  Status stat((StandardType<typename Packing<T>::buffer_type>()));
+
+  int int_flag;
+
+  libmesh_call_mpi(MPI_Iprobe(src_processor_id,
+                              tag.value(),
+                              this->get(),
+                              &int_flag,
+                              stat.get()));
+
+  flag = int_flag;
+
+  STOP_LOG("packed_range_probe()", "Parallel");
+
+  return stat;
+}
 
```

Comment:
Oh - one more thing.  I see that in a lot of the existing functions we're passing a pointer through just for the purpose of defining the template type.  Is there a reason we do that instead of just requiring template arguments on the function?  For instance, the above code could look like:

```C++
if (comm.nonblocking_probe(0, my_tag, my_status))
  comm.nonblocking_receive_packed_range<MyType>(my_context, my_status);
```